File: onyx-database-tests/test/remote/partition/IndexPartitionTest.java
package remote.partition;

import category.RemoteServerTests;
import com.onyx.exception.EntityException;
import com.onyx.persistence.query.Query;
import com.onyx.persistence.query.QueryCriteria;
import com.onyx.persistence.query.QueryCriteriaOperator;
import com.onyx.persistence.update.AttributeUpdate;
import entities.partition.IndexPartitionEntity;
import org.junit.Assert;
import org.junit.FixMethodOrder;
import org.junit.Test;
import org.junit.experimental.categories.Category;
import org.junit.runners.MethodSorters;

import java.util.Arrays;
import java.util.List;

/**
 * Created by timothy.osborn on 2/12/15.
 */
@Category({ RemoteServerTests.class })
@FixMethodOrder(MethodSorters.NAME_ASCENDING)
public class IndexPartitionTest extends BasePartitionTest
{
    @Test
    public void aTestSavePartitionEntityWithIndex()
    {
        IndexPartitionEntity IndexPartitionEntity = new IndexPartitionEntity();
        IndexPartitionEntity.id = 1l;
        IndexPartitionEntity.partitionId = 3l;
        IndexPartitionEntity.indexVal = 5l;

        save(IndexPartitionEntity);
    }

    @Test
    public void bTestQueryPartitionEntityWithIndex() throws EntityException
    {
        IndexPartitionEntity IndexPartitionEntity = new IndexPartitionEntity();
        IndexPartitionEntity.id = 1l;
        IndexPartitionEntity.partitionId = 3l;
        IndexPartitionEntity.indexVal = 5l;

        save(IndexPartitionEntity);

        entities.partition.IndexPartitionEntity IndexPartitionEntity2 = new IndexPartitionEntity();
        IndexPartitionEntity2.id = 2l;
        IndexPartitionEntity2.partitionId = 2l;
        IndexPartitionEntity2.indexVal = 5l;

        save(IndexPartitionEntity2);

        Query query = new Query(entities.partition.IndexPartitionEntity.class, new QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 5l));
        query.setPartition(2l);

        List results = manager.executeQuery(query);
        Assert.assertTrue(results.size() == 1);
    }

    @Test
    public void bTestQueryPartitionEntityWithIndexNoDefinedPartitionInQuery() throws EntityException
    {
        IndexPartitionEntity IndexPartitionEntity = new IndexPartitionEntity();
        IndexPartitionEntity.id = 1l;
        IndexPartitionEntity.partitionId = 3l;
        IndexPartitionEntity.indexVal = 5l;

        save(IndexPartitionEntity);

        entities.partition.IndexPartitionEntity IndexPartitionEntity2 = new IndexPartitionEntity();
        IndexPartitionEntity2.id = 2l;
        IndexPartitionEntity2.partitionId = 2l;
        IndexPartitionEntity2.indexVal = 5l;

        save(IndexPartitionEntity2);

        Query query = new Query(entities.partition.IndexPartitionEntity.class, new QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 5l));

        List results = manager.executeQuery(query);
        Assert.assertTrue(results.size() == 2);
    }

    @Test
    public void cTestQueryFindQueryPartitionEntityWithIndex() throws EntityException
    {
        IndexPartitionEntity IndexPartitionEntity = new IndexPartitionEntity();
        IndexPartitionEntity.id = 1l;
        IndexPartitionEntity.partitionId = 3l;
        IndexPartitionEntity.indexVal = 5l;

        save(IndexPartitionEntity);

        entities.partition.IndexPartitionEntity IndexPartitionEntity2 = new IndexPartitionEntity();
        IndexPartitionEntity2.id = 2l;
        IndexPartitionEntity2.partitionId = 2l;
        IndexPartitionEntity2.indexVal = 5l;

        save(IndexPartitionEntity2);

        Query query = new Query(entities.partition.IndexPartitionEntity.class, new QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 5l).and("partitionId", QueryCriteriaOperator.EQUAL, 2l));

        List results = manager.executeQuery(query);
        Assert.assertTrue(results.size() == 1);
    }

    @Test
    public void dTestDeleteQueryPartitionEntity() throws EntityException
    {
        IndexPartitionEntity IndexPartitionEntity = new IndexPartitionEntity();
        IndexPartitionEntity.id = 1l;
        IndexPartitionEntity.partitionId = 3l;
        IndexPartitionEntity.indexVal = 5l;

        save(IndexPartitionEntity);

        entities.partition.IndexPartitionEntity IndexPartitionEntity2 = new IndexPartitionEntity();
        IndexPartitionEntity2.id = 2l;
        IndexPartitionEntity2.partitionId = 2l;
        IndexPartitionEntity2.indexVal = 5l;

        save(IndexPartitionEntity2);

        Query query = new Query(entities.partition.IndexPartitionEntity.class, new QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 5l).and("partitionId", QueryCriteriaOperator.EQUAL, 2l));

        int results = manager.executeDelete(query);
        Assert.assertTrue(results == 1);
    }

    @Test
    public void bTestDeleteQueryPartitionEntityWithIndex() throws EntityException
    {
        IndexPartitionEntity IndexPartitionEntity = new IndexPartitionEntity();
        IndexPartitionEntity.id = 1l;
        IndexPartitionEntity.partitionId = 3l;
        IndexPartitionEntity.indexVal = 5l;

        save(IndexPartitionEntity);

        entities.partition.IndexPartitionEntity IndexPartitionEntity2 = new IndexPartitionEntity();
        IndexPartitionEntity2.id = 2l;
        IndexPartitionEntity2.partitionId = 2l;
        IndexPartitionEntity2.indexVal = 5l;

        save(IndexPartitionEntity2);

        Query query = new Query(entities.partition.IndexPartitionEntity.class, new QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 5l));
        query.setPartition(2l);

        int results = manager.executeDelete(query);
        Assert.assertTrue(results == 1);
    }

    @Test
    public void bTestDeleteQueryPartitionEntityWithIndexNoDefinedPartitionInQuery() throws EntityException
    {
        IndexPartitionEntity IndexPartitionEntity = new IndexPartitionEntity();
        IndexPartitionEntity.id = 1l;
        IndexPartitionEntity.partitionId = 3l;
        IndexPartitionEntity.indexVal = 5l;

        save(IndexPartitionEntity);

        entities.partition.IndexPartitionEntity IndexPartitionEntity2 = new IndexPartitionEntity();
        IndexPartitionEntity2.id = 2l;
        IndexPartitionEntity2.partitionId = 2l;
        IndexPartitionEntity2.indexVal = 5l;

        save(IndexPartitionEntity2);

        Query query = new Query(entities.partition.IndexPartitionEntity.class, new QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 5l));

        int results = manager.executeDelete(query);
        Assert.assertTrue(results == 2);
    }


    @Test
    public void dTestUpdateQueryPartitionEntity() throws EntityException
    {
        IndexPartitionEntity IndexPartitionEntity = new IndexPartitionEntity();
        IndexPartitionEntity.id = 1l;
        IndexPartitionEntity.partitionId = 3l;
        IndexPartitionEntity.indexVal = 5l;

        save(IndexPartitionEntity);

        entities.partition.IndexPartitionEntity IndexPartitionEntity2 = new IndexPartitionEntity();
        IndexPartitionEntity2.id = 2l;
        IndexPartitionEntity2.partitionId = 2l;
        IndexPartitionEntity2.indexVal = 5l;

        save(IndexPartitionEntity2);

        Query query = new Query(entities.partition.IndexPartitionEntity.class, new QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 5l).and("partitionId", QueryCriteriaOperator.EQUAL, 2l));
        query.setUpdates(Arrays.asList(new AttributeUpdate("indexVal", 6l)));

        int results = manager.executeUpdate(query);
        Assert.assertTrue(results == 1);
    }

    @Test
    public void bTestUpdateQueryPartitionEntityWithIndex() throws EntityException
    {
        IndexPartitionEntity IndexPartitionEntity = new IndexPartitionEntity();
        IndexPartitionEntity.id = 1l;
        IndexPartitionEntity.partitionId = 3l;
        IndexPartitionEntity.indexVal = 5l;

        save(IndexPartitionEntity);

        entities.partition.IndexPartitionEntity IndexPartitionEntity2 = new IndexPartitionEntity();
        IndexPartitionEntity2.id = 2l;
        IndexPartitionEntity2.partitionId = 2l;
        IndexPartitionEntity2.indexVal = 5l;

        save(IndexPartitionEntity2);

        Query query = new Query(entities.partition.IndexPartitionEntity.class, new QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 5l));
        query.setUpdates(Arrays.asList(new AttributeUpdate("indexVal", 6l)));
        query.setPartition(2l);

        int results = manager.executeUpdate(query);
        Assert.assertTrue(results == 1);
    }

    @Test
    public void bTestUpdateQueryPartitionEntityWithIndexNoDefinedPartitionInQuery() throws EntityException
    {
        IndexPartitionEntity IndexPartitionEntity = new IndexPartitionEntity();
        IndexPartitionEntity.id = 1l;
        IndexPartitionEntity.partitionId = 3l;
        IndexPartitionEntity.indexVal = 5l;

        save(IndexPartitionEntity);

        entities.partition.IndexPartitionEntity IndexPartitionEntity2 = new IndexPartitionEntity();
        IndexPartitionEntity2.id = 2l;
        IndexPartitionEntity2.partitionId = 2l;
        IndexPartitionEntity2.indexVal = 5l;

        save(IndexPartitionEntity2);

        Query query = new Query(entities.partition.IndexPartitionEntity.class, new QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 5l));
        query.setUpdates(Arrays.asList(new AttributeUpdate("indexVal", 6l)));

        int results = manager.executeUpdate(query);
        Assert.assertTrue(results == 2);
    }

    @Test
    public void bTestUpdatePartitionField() throws EntityException
    {
        IndexPartitionEntity IndexPartitionEntity = new IndexPartitionEntity();
        IndexPartitionEntity.id = 1l;
        IndexPartitionEntity.partitionId = 3l;
        IndexPartitionEntity.indexVal = 5l;

        save(IndexPartitionEntity);

        entities.partition.IndexPartitionEntity IndexPartitionEntity2 = new IndexPartitionEntity();
        IndexPartitionEntity2.id = 2l;
        IndexPartitionEntity2.partitionId = 2l;
        IndexPartitionEntity2.indexVal = 5l;

        save(IndexPartitionEntity2);

        Query query = new Query(entities.partition.IndexPartitionEntity.class, new QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 5l));
        query.setUpdates(Arrays.asList(new AttributeUpdate("partitionId", 5l), new AttributeUpdate("indexVal", 6l)));

        int results = manager.executeUpdate(query);
        Assert.assertTrue(results == 2);

        Query query2 = new Query(entities.partition.IndexPartitionEntity.class, new QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 6l));
        query2.setPartition(5l);
        List result = manager.executeQuery(query2);

        Assert.assertTrue(result.size() == 2);
    }

}

File: onyx-database-tests/test/remote/partition/IndexPartitionTest.kt
package remote.partition

import category.RemoteServerTests
import com.onyx.exception.OnyxException
import com.onyx.persistence.query.Query
import com.onyx.persistence.query.QueryCriteria
import com.onyx.persistence.query.QueryCriteriaOperator
import com.onyx.persistence.query.AttributeUpdate
import entities.partition.IndexPartitionEntity
import org.junit.Assert
import org.junit.FixMethodOrder
import org.junit.Test
import org.junit.experimental.categories.Category
import org.junit.runners.MethodSorters

import java.util.Arrays

/**
 * Created by timothy.osborn on 2/12/15.
 */
@Category(RemoteServerTests::class)
@FixMethodOrder(MethodSorters.NAME_ASCENDING)
class IndexPartitionTest : BasePartitionTest() {

    @Test
    fun aTestSavePartitionEntityWithIndex() {
        val IndexPartitionEntity = IndexPartitionEntity()
        IndexPartitionEntity.id = 1L
        IndexPartitionEntity.partitionId = 3L
        IndexPartitionEntity.indexVal = 5L

        save(IndexPartitionEntity)
    }

    @Test
    @Throws(OnyxException::class)
    fun bTestQueryPartitionEntityWithIndex() {
        val IndexPartitionEntity = IndexPartitionEntity()
        IndexPartitionEntity.id = 1L
        IndexPartitionEntity.partitionId = 3L
        IndexPartitionEntity.indexVal = 5L

        save(IndexPartitionEntity)

        val IndexPartitionEntity2 = IndexPartitionEntity()
        IndexPartitionEntity2.id = 2L
        IndexPartitionEntity2.partitionId = 2L
        IndexPartitionEntity2.indexVal = 5L

        save(IndexPartitionEntity2)

        val query = Query(entities.partition.IndexPartitionEntity::class.java, QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 5L))
        query.partition = 2L

        val results = manager!!.executeQuery<Any>(query)
        Assert.assertTrue(results.size == 1)
    }

    @Test
    @Throws(OnyxException::class)
    fun cTestQueryPartitionEntityWithIndexNoDefinedPartitionInQuery() {
        val IndexPartitionEntity = IndexPartitionEntity()
        IndexPartitionEntity.id = 1L
        IndexPartitionEntity.partitionId = 3L
        IndexPartitionEntity.indexVal = 5L

        save(IndexPartitionEntity)

        val IndexPartitionEntity2 = IndexPartitionEntity()
        IndexPartitionEntity2.id = 2L
        IndexPartitionEntity2.partitionId = 2L
        IndexPartitionEntity2.indexVal = 5L

        save(IndexPartitionEntity2)

        val query = Query(entities.partition.IndexPartitionEntity::class.java, QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 5L))

        val results = manager!!.executeQuery<Any>(query)
        Assert.assertTrue(results.size == 2)
    }

    @Test
    @Throws(OnyxException::class)
    fun dTestQueryFindQueryPartitionEntityWithIndex() {
        val IndexPartitionEntity = IndexPartitionEntity()
        IndexPartitionEntity.id = 1L
        IndexPartitionEntity.partitionId = 3L
        IndexPartitionEntity.indexVal = 5L

        save(IndexPartitionEntity)

        val IndexPartitionEntity2 = IndexPartitionEntity()
        IndexPartitionEntity2.id = 2L
        IndexPartitionEntity2.partitionId = 2L
        IndexPartitionEntity2.indexVal = 5L

        save(IndexPartitionEntity2)

        val query = Query(entities.partition.IndexPartitionEntity::class.java, QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 5L).and("partitionId", QueryCriteriaOperator.EQUAL, 2L))

        val results = manager!!.executeQuery<Any>(query)
        Assert.assertTrue(results.size == 1)
    }

    @Test
    @Throws(OnyxException::class)
    fun eTestDeleteQueryPartitionEntity() {
        val IndexPartitionEntity = IndexPartitionEntity()
        IndexPartitionEntity.id = 1L
        IndexPartitionEntity.partitionId = 3L
        IndexPartitionEntity.indexVal = 5L

        save(IndexPartitionEntity)

        val IndexPartitionEntity2 = IndexPartitionEntity()
        IndexPartitionEntity2.id = 2L
        IndexPartitionEntity2.partitionId = 2L
        IndexPartitionEntity2.indexVal = 5L

        save(IndexPartitionEntity2)

        val query = Query(entities.partition.IndexPartitionEntity::class.java, QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 5L).and("partitionId", QueryCriteriaOperator.EQUAL, 2L))

        val results = manager!!.executeDelete(query)
        Assert.assertTrue(results == 1)
    }

    @Test
    @Throws(OnyxException::class)
    fun fTestDeleteQueryPartitionEntityWithIndex() {
        val IndexPartitionEntity = IndexPartitionEntity()
        IndexPartitionEntity.id = 1L
        IndexPartitionEntity.partitionId = 3L
        IndexPartitionEntity.indexVal = 5L

        save(IndexPartitionEntity)

        val IndexPartitionEntity2 = IndexPartitionEntity()
        IndexPartitionEntity2.id = 2L
        IndexPartitionEntity2.partitionId = 2L
        IndexPartitionEntity2.indexVal = 5L

        save(IndexPartitionEntity2)

        val query = Query(entities.partition.IndexPartitionEntity::class.java, QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 5L))
        query.partition = 2L

        val results = manager!!.executeDelete(query)
        Assert.assertTrue(results == 1)
    }

    @Test
    @Throws(OnyxException::class)
    fun gTestDeleteQueryPartitionEntityWithIndexNoDefinedPartitionInQuery() {
        val IndexPartitionEntity = IndexPartitionEntity()
        IndexPartitionEntity.id = 1L
        IndexPartitionEntity.partitionId = 3L
        IndexPartitionEntity.indexVal = 5L

        save(IndexPartitionEntity)

        val IndexPartitionEntity2 = IndexPartitionEntity()
        IndexPartitionEntity2.id = 2L
        IndexPartitionEntity2.partitionId = 2L
        IndexPartitionEntity2.indexVal = 5L

        save(IndexPartitionEntity2)

        val query = Query(entities.partition.IndexPartitionEntity::class.java, QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 5L))

        val results = manager!!.executeDelete(query)
        Assert.assertTrue(results == 2)
    }


    @Test
    @Throws(OnyxException::class)
    fun hTestUpdateQueryPartitionEntity() {
        val IndexPartitionEntity = IndexPartitionEntity()
        IndexPartitionEntity.id = 1L
        IndexPartitionEntity.partitionId = 3L
        IndexPartitionEntity.indexVal = 5L

        save(IndexPartitionEntity)

        val IndexPartitionEntity2 = IndexPartitionEntity()
        IndexPartitionEntity2.id = 2L
        IndexPartitionEntity2.partitionId = 2L
        IndexPartitionEntity2.indexVal = 5L

        save(IndexPartitionEntity2)

        val query = Query(entities.partition.IndexPartitionEntity::class.java, QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 5L).and("partitionId", QueryCriteriaOperator.EQUAL, 2L))
        query.updates = Arrays.asList(AttributeUpdate("indexVal", 6L))

        val results = manager!!.executeUpdate(query)
        Assert.assertTrue(results == 1)
    }

    @Test
    @Throws(OnyxException::class)
    fun iTestUpdateQueryPartitionEntityWithIndex() {
        val IndexPartitionEntity = IndexPartitionEntity()
        IndexPartitionEntity.id = 1L
        IndexPartitionEntity.partitionId = 3L
        IndexPartitionEntity.indexVal = 5L

        save(IndexPartitionEntity)

        val IndexPartitionEntity2 = IndexPartitionEntity()
        IndexPartitionEntity2.id = 2L
        IndexPartitionEntity2.partitionId = 2L
        IndexPartitionEntity2.indexVal = 5L

        save(IndexPartitionEntity2)

        val query = Query(entities.partition.IndexPartitionEntity::class.java, QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 5L))
        query.updates = Arrays.asList(AttributeUpdate("indexVal", 6L))
        query.partition = 2L

        val results = manager!!.executeUpdate(query)
        Assert.assertTrue(results == 1)
    }

    @Test
    @Throws(OnyxException::class)
    fun jTestUpdateQueryPartitionEntityWithIndexNoDefinedPartitionInQuery() {
        val IndexPartitionEntity = IndexPartitionEntity()
        IndexPartitionEntity.id = 1L
        IndexPartitionEntity.partitionId = 3L
        IndexPartitionEntity.indexVal = 5L

        save(IndexPartitionEntity)

        val IndexPartitionEntity2 = IndexPartitionEntity()
        IndexPartitionEntity2.id = 2L
        IndexPartitionEntity2.partitionId = 2L
        IndexPartitionEntity2.indexVal = 5L

        save(IndexPartitionEntity2)

        val query = Query(entities.partition.IndexPartitionEntity::class.java, QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 5L))
        query.updates = Arrays.asList(AttributeUpdate("indexVal", 6L))

        val results = manager!!.executeUpdate(query)
        Assert.assertTrue(results == 2)
    }

    @Test
    @Throws(OnyxException::class)
    fun kTestUpdatePartitionField() {
        val IndexPartitionEntity = IndexPartitionEntity()
        IndexPartitionEntity.id = 1L
        IndexPartitionEntity.partitionId = 3L
        IndexPartitionEntity.indexVal = 5L

        save(IndexPartitionEntity)

        val IndexPartitionEntity2 = IndexPartitionEntity()
        IndexPartitionEntity2.id = 2L
        IndexPartitionEntity2.partitionId = 2L
        IndexPartitionEntity2.indexVal = 5L

        save(IndexPartitionEntity2)

        val query = Query(entities.partition.IndexPartitionEntity::class.java, QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 5L))
        query.updates = Arrays.asList(AttributeUpdate("partitionId", 5L), AttributeUpdate("indexVal", 6L))

        val results = manager!!.executeUpdate(query)
        Assert.assertTrue(results == 2)

        val query2 = Query(entities.partition.IndexPartitionEntity::class.java, QueryCriteria("indexVal", QueryCriteriaOperator.EQUAL, 6L))
        query2.partition = 5L
        val result = manager!!.executeQuery<Any>(query2)

        Assert.assertTrue(result.size == 2)
    }

}